### DISCUSSION TITLE
INVERTER not supported

### DISCUSSION THREAD

**Discussion by balbright05** [CharlesGodwin/pymagnum#58] [General]
Hello, 
Thanks for all the work you have done. I have stumbled upon pymagnum in my pursuit to gather data from my MS4448PAE. In my current configuration I have a single inverter and an ARTR connected. When running magsample.py It only finds the REMOTE. Using magtest I'm getting an INVERTER not supported. I have the RJ11 splitter directly on the inverter remote port and split to my RP USB 485 adapter and ARTR. Any suggestions?

>>> %Run magsample.py
[
  {
    "device": "REMOTE",
    "data": {
      "revision": "25.5",
      "searchwatts": 175,
      "batterysize": 2430,
      "battype": 0,
      "absorb": 13.5,
      "chargeramps": 175,
      "ainput": -51,
      "parallel": 150,
      "lbco": 24.3,
      "vaccutout": 153.0
    }
  }
]

admin@raspberrypi4:~ $ magtest --device /dev/ttyUSB0
Magnum Test Version:2.0.5
Options:device='/dev/ttyUSB0', packets=50, timeout=0.005, trace=False, cleanpackets=True, log=False
Length:16 REMOTE_C  =>FFAFF387AFCDFF1F90F3993FFF1E0F0F
Length:39 UNKNOWN   =>FDFBA9FFFF17FFFDFF99CDB1BF19FFFFFFFB4FFF01FFAFF387AFCDFF1F90F399FFFFFFFFFFFF5F
Length:39 UNKNOWN   =>FDFBA9FFFF17FFFDFF99CDB1BF19FFFFFFFB4FFF01FFAFF387AFCDFF1F90F399FFFFFFEFFFFF7F
Length:40 UNKNOWN   =>FDFBA7FFFF17FFFDFF99CDB1BF19FFFFFFFB4FFF01FFAFF387AFCDFF1F90F399FFB9FFFFFFFFFF00
Length:39 UNKNOWN   =>FDFBA7FFFF17FFFDFF99CDB1BF19FFFFFFFB4FFF01FFAFF387AFCDFF1F90F399FFB9FFFFAFFF00
Length:39 UNKNOWN   =>FDFBA9FFFF17FFFDFF99CDB1BF19FFFFFFFB4FFF01FFAFF387AFCDFF1F90F399FFB9D7FF19FF01
Length:39 UNKNOWN   =>FDFBA9FFFF17FFFDFF99CDB1BF19FFFFFFFB4FFF03FFAFF387AFCDFF1F90F399D73CFFFFFFDD00
Length:16 REMOTE_C  =>FFAFF387AFCDFF1F90F3993FFF1E0F0F
Length:39 UNKNOWN   =>FDFBA7FFFF17FFFDFF99CDB1BF19FFFFFFFB4FFF01FFAFF387AFCDFF1F90F399FFFFFFFFFFFF5F
Length:39 UNKNOWN   =>FDFBA7FFFF17FFFDFF99CDB1BF19FFFFFFFB4FFF01FFAFF387AFCDFF1F90F399FFFFFFDFFFFF7F
Length:40 UNKNOWN   =>FDFBA7FFFF17FFFDFF99CDB1BF19FFFFFFFB4FFF07FFAFF387AFCDFF1F90F399FFB9FFFFFFFFFF00
Length:39 UNKNOWN   =>FDFBA7FFFF17FFFDFF99CDB1BF19FFFFFFFB4FFF03FFAFF387AFCDFF1F90F399FFB9FFFFAFFF00
Length:39 UNKNOWN   =>FDFBA9FFFF17FFFDFF99CDB1BF19FFFFFFFB4FFF01FFAFF387AFCDFF1F90F399FFB9D7FF19FF01
Length:39 UNKNOWN   =>FDFBA7FFFF1BFFFDFF99CDB1BF19FFFFFFFB4FFF03FFAFF387AFCDFF1F90F399D73CFFFFFFDD00
Length:16 REMOTE_C  =>FFAFF387AFCDFF1F90F3993FFF1E0F0F
Length:39 UNKNOWN   =>FDFBA9FFFF1BFFFDFF99CDB1BF19FFFFFFFB4FFF01FFAFF387AFCDFF1F90F399FFFFFFFFFFFF5F
Length:39 UNKNOWN   =>FDFBA9FFFF1BFFFDFF99CDB1BF19FFFFFFFB4FFF03FFAFF387AFCDFF1F90F399FFFFFFEFFFFF7F
Length:40 UNKNOWN   =>FDFBA7FFFF1BFFFDFF99CDB1BF19FFFFFFFB4FFF01FFAFF387AFCDFF1F90F399FFB9FFFFFFFFFF00
Length:39 UNKNOWN   =>FDFBA7FFFF1BFFFDFF99CDB1BF19FFFFFFFB4FFF01FFAFF387AFCDFF1F90F399FFB9FFFFAFFF00
Length:39 UNKNOWN   =>FDFBA7FFFF1BFFFDFF99CDB1BF19FFFFFFFB4FFF01FFAFF387AFCDFF1F90F399FFB9D7FF19FF01
Length:39 UNKNOWN   =>FDFBA7FFFF1BFFFDFF99CDB1BF19FFFFFFFB4FFF01FFAFF387AFCDFF1F90F399D73CFFFFFFDD00
Length:16 REMOTE_C  =>FFAFF387AFCDFF1F90F3993FFF1E0F0F
Length:39 UNKNOWN   =>FDFBA9FFFF19FFFDFF99CDB1BF19FFFFFFFB4FFF01FFAFF387AFCDFF1F90F399FFFFFFFFFFFF5F
Length:39 UNKNOWN   =>FDFBA9FFFF19FFFDFF99CDB1BF19FFFFFFFB4FFF07FFAFF387AFCDFF1F90F399FFFFFFDFFFFF7F
Length:40 UNKNOWN   =>FDFBA9FFFF19FFFDFF99CDB1BF19FFFFFFFB4FFF01FFAFF387AFCDFF1F90F399FFB9FFFFFFFFFF00
Packets:25 of 50 with 21 UNKNOWN, in 3.16 seconds
INVERTER not supported
REMOTE Detected
RTR not connected
BMK not supported
AGS not supported
PT100 not supported
ACLD not supported

**Comment by CharlesGodwin** [comment]
Try reversing the wire to your 4S485 adapter/ dongle. You have a high rate of UNKNOWN which is one sign of reversed wires. Alas, I don't know a way to automatically reverse the wiring 8n software

**Reply by balbright05** [reply]
That appears to have been the issue. I thought I had tried that combination. Thanks for the quick response.

**Reply by CharlesGodwin** [reply]
I'm delighted it was easy to resolve
